server/src/main/java/io/littlehorse/common/model/getable/core/taskrun/TaskAttemptModel.java

```diff
@@ -45,8 +47,13 @@ public Class<TaskAttempt> getProtoBaseClass() {
     @Override
     public void initFrom(Message proto, ExecutionContext context) {
         TaskAttempt p = (TaskAttempt) proto;
+        maskedValue = p.getMaskedValue();
         if (p.hasOutput()) {
-            output = VariableValueModel.fromProto(p.getOutput(), context);
+            if (maskedValue && context instanceof RequestExecutionContext) {
+                output = new VariableValueModel(LHConstants.STRING_MASK);
```

Comment:
The only concern I have here is that non- `STR` variables will be converted into `STR`.

Reply:
It makes sense to me because mask is a String literal.